torch/csrc/jit/python/pybind_utils.h

```diff
@@ -286,6 +286,18 @@ inline InferredType tryToInferType(py::handle input) {
     return InferredType(IntType::get());
   }
 
+  auto enum_type = py::module::import("enum").attr("Enum");
+  py::bool_ isEnumValue = py::isinstance(input, enum_type);
+  if (py::cast<bool>(isEnumValue)) {
+    auto enum_class = input.attr("__class__");
+    py::str qualified_name = py::module::import("torch._jit_internal")
+                                 .attr("_qualified_name")(enum_class);
+    auto qualname = c10::QualifiedName(qualified_name);
+    auto value_type = tryToInferType(input.attr("value")).type();
```

Comment:
I think this has the same problems as we did previously,  where we're individually introspecting the object's value so the EnumType doesn't get registered and  we won't hit a compile error if someone tries to use an Enum with heterogenous types.

Reply:
Done. I had to pass a py::none() as location to try_ann_to_type since the info isn't available here though.

Reply:
Done.